client/components/Index/SignUpModal.jsx

```diff
@@ -18,24 +24,33 @@ class SignUpModal extends React.Component {
     this.onChange = this.onChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
   }
+  /**
+   * @return {null} Triggers the signupRequest action on submit button click
+   * @param {e} e
+   */
   onSubmit(e) {
     this.setState({ errors: {} });
     e.preventDefault();
     this.props.signupRequest(this.state).then(
       () => {
 
       },
-      (errors) => {
-        console.log(errors, '============');
-        debugger;
+      () => {
       }
     );
   }
+  /**
+   * @return {null} Updates the state as the user types into the input field
+   * @param {e} e
+   */
   onChange(e) {
     const state = this.state;
```

Comment:
Same issue I pointed out earlier, you're not making a copy of state, you're only storing a pointer.